hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -328,24 +328,54 @@ public StreamSync(HoodieStreamer.Config cfg, SparkSession sparkSession,
   }
 
   /**
-   * Refresh Timeline.
+   * Creates a meta client for the table, and refresh timeline.
    *
    * @throws IOException in case of any IOException
    */
-  public void refreshTimeline() throws IOException {
+  public HoodieTableMetaClient initializeMetaClientAndRefreshTimeline() throws IOException {
```

Comment:
We can just throw `HoodieIOException` instead of `IOException`

Reply:
I was trying not to change the signatures too much here